# Dissertation-

SupportingTable:

1:Supporting Materials A: Orginal phenotype data

2:Supporting Materials B: The summarized GWAS results of nine traits

3:Supporting Materials C: The GWAS results comparison between SNPs from current study and QTLs and SNPs from other studies.

4:Supporting Materials D: The known flowering genes from other species and miscanthus.

5:Supporting Materials E: The summarized GWAS results of flowering traits.

6:Supporting Materials F: The genes related to cold and overwintering ability from other species and Miscanthus.

7:Supporting Materials G:The summarized GWAS results of Overwintering Ability.

8:Supporting Materials H:The GWAS results of all traits.


Data: including several datasets used for the analysis. 

OWA: including datasets of OWA genes from other species. 

OWA2: including datasets used for making the table and images for the traits of OWA

CMLMSUPERCV0.05:GWAS results from CMLM joint with SUPER with PS

CMLMSUPER0.05:GWAS results from CMLM joint with SUPER without PS

ECMMLCV0.05: GWAS results from ECMLM with PS.

ECMML0.05:GWAS results from ECMLM without PS

FarmCPUCV0.05:GWAS results from FarmCPU with PS

FarmCPU0.05:GWAS results from FarmCPU without PS

GLMMLMMLMMCV0.05:GWAS results from GLM MLM MLMM with PS

GLMMLMMLMM0.05:GWAS results from GLM MLM MLMM without PS

MLMSUPERCV0.05:GWAS results from MLM joint with SUPER with PS

MLMSUPER0.05:GWAS results from MLM joint with SUPER without PS

rrBLUPim:GWAS results of all traits using imputed SNPs datasets using rrBLUP software

rrBLUPF96: GWAS results of flowering traits using F96 SNPs datasets using rrBLUP software

rrBLUPF106:GWAS results of flowering traits using F96 SNPs datasets using rrBLUP software

rrBLUPF116:GWAS results of flowering traits using F96 SNPs datasets using rrBLUP software

rrBLUPC106:GWAS results of culm related traits using C106 SNPs datasets using rrBLUP software

rrBLUPC116:GWAS results of culm related traits using C116 SNPs datasets using rrBLUP software

rrBLUPC124:GWAS results of culm related traits using C124 SNPs datasets using rrBLUP software

rrBLUPSRD:GWAS results of SRD using all SNPs datasets using rrBLUP software

mrMLMM2: GWAS results of all traits using all SNPs datasets using all the GWAS methods in mrMLM R package

Chapter2UP: all datasets used for the table and images for nine traits

Flotraits2: all datasets used for the table and images for the flowering traits


Rcode includes all of the R codes used in the dissertation 

0 Mis1718-checking outliers: checking phenotype data outliers

1 Mis-descriptive statistics: producing descriptive statsitics

2 Mis-Ftpc1with PPCA: obtaining the PCA of the flowering traits

3 Mis-Heritability and BLUP: obtaining the heritability and Best Linear Unbiased Predictors

4 Mis-Lsmean vs Blup-plot: the plot of Lsmean and BLUP

5 Mis-imputedSNP-19: preparing the imputed SNPs datasets for GWAS methods in GAPIT R package

5-1 MisClu-NONMisingSNP:preparing the three non missing SNPs datasets for culm related traits used for GWAS methods in GAPIT R package.

5-2 MisFlo-NONMisingSNP:preparing the imputed SNPs datasets for flowering traits used for GWAS methods in GAPIT R package.

5-3 MisOWA-NONMisingSNP:preparing the imputed SNPs datasets for overwintering ability used for GWAS methods in GAPIT R package.

6-1 GAPIT:GWAS analysis using all the methods in GAPIT R package.

6-2 FarmCPU:FarmCPU method

7-0 ImputedRR-BLUP-gwas:The imputed SNPs dataset preparation for rrBLUP R package 

7-1 ClumNonmissing RR-BLUP-gwas:The three non missing SNPs dataset preparation for culm traits used for rrBLUP R package

7-2 FloNonmissing RR-BLUP-gwas:The three non missing SNPs dataset preparation for flowering traits used for rrBLUP R package

7-3 OWANonmissing RR-BLUP-gwas:The three non missing SNPs dataset preparation for overwintering ability used for rrBLUP R package

8-1 FloCulmOWAImputed-mrMLM: The imputed SNPs datasets preparation for all GWAS methods in mrMLM R package.

8-2 FloClumOWANonMissing-mrMLM: The three non missing SNPs datasets preparation for all GWAS methods in mrMLM R package.

9 all Results from GWAS: summary all GWAS results from all different R packages

10-1 mul-imag-F1-CV vs NOCV: QQ and manhattan plots with PS and without PS comparison in method FarmCPU and SUPER

10-1 multipl image flo:QQ and Manhattan plots for all the flowering traits.

11-1 PVE.FarmCPUCV: calculating PVE for method FarmCPU with PS

11-2 PVE.FarmCPUNOCV: calculating PVE for method FarmCPU without PS

11-3 PVE.MLMSUPER:calculating PVE for method MLMSUPER with PS

11-4 PVE.CMLMSUPER:calculating PVE for method CMLMSUPER with PS

11-5 PVE.rrBLUP:calculating PVE for method rrBLUP with PS

12 MAF and PVE for all traits: combine MAF and PVE with the GWAS results of all traits

13-1 Flo Gene Location:find the nearby genes for flowering traits

13-2 OWA Gene Location:find the nearby genes for overwintering ability trait

13-3 Nine Trait Gene Location:find the nearby genes for nine traits

14-1 Flo tables and images:used for making tables and images for the flowering trait.

14-2 Ninetrait tables and images:used for making tables and images for the nine traits.

14-3 OWA tables and images:used for making tables and images for the OWA traits.

15 Draw GWAS Comparison Chromosome: used for drawing chromsome labels SNPs from current study and QTLs or SNPs from other studies.

16-1 GPKin_blup_NOPC:Genomic prediction without PS

16-2 GPKin_Blup_PC1:Genomic prediction with PC1

16-3 GPKin_Blup_PC2:Genomic prediction with PC2

16-4 GPKin_blup_PC3:Genomic prediction with PC3

17 Mean and Varance of GP: making images used for the mean and standand derivation genomic prediciton

Function: including all the function used for the Rcodes 





